### PR TITLE
Rating system validation

### LINKED ISSUE #86: Rating system validation
Implement validation on server side to verify if a given Rating System is created with valid parameters
- [x] Validation checks for ALL rating systems
    - [x] At least one subrating
    - [x] All subratings weights add up to 1.0 (within some error because floating point !!)
    - [x] Size of at least 2
    - [x] A non null name of at most 50 characters
    - [x] Name is alphanumeric
- [x] Validation checks for continuous
    - [x] offset >= 0 (negative offsets might be interesting tho lol)
- [x] Validation checks for discrete
    - [x] amount of labels == size
    - [x] labels are non null/ non empty
- [x] Appropriate unit tests

### PR DESCRIPTION
**Pull request by jvoucsb**
## Notes
- Add Rating System validation according to acceptance criteria
- Fixed unit tests for Creating Rating system to fit in new criteria
## Testing
- By unit tests to make sure the assertions are working correctly
Closes #86

### CODE REVIEW

**Comment by jvoucsb** [comment]
oops realized names must be alphanumeric and contain space and hypens too

**Review by AHuangHe — APPROVED** [review]
LGTM!

File: server/src/main/java/com/github/animelist/animelist/model/ratingsystem/RatingSystem.java [hunk 1]
```diff
@@ -23,12 +26,24 @@
...
         this.name = name;
         this.ownerId = ownerId;
         this.size = size;
         this.subRatings = subRatings;
+        Assert.isTrue(size >= 2, "Size must be at least 2");
+        Assert.isTrue(subRatings.size()>=1, "subRatings size must be at least 1");
+        Assert.isTrue(nonNull(name) && !name.isBlank(),"name cannot be blank");
+        Assert.isTrue(name.length()>=1 && name.length()<= 50, "name must be at least one character and less than 50 characters");
+        var weightSum = subRatings.stream().mapToDouble(SubRating::getWeight).sum();
+        var difference = WEIGHT_SUM_EXPECTED - weightSum;
```
Comment (rowantran) [inline server/src/main/java/com/github/animelist/animelist/model/ratingsystem/RatingSystem.java:44]: the floating point error could be in either direction so I think we should take the absolute value of the difference here? then we could also get rid of the >=0 check on line 45

**Review by rowantran — CHANGES_REQUESTED** [review]
left a comment regarding the use of epsilon for checking the sum of subrating weights. other than that everything looks good to me, good work!

File: server/src/main/java/com/github/animelist/animelist/model/ratingsystem/RatingSystem.java (hunk 1 above)
Comment (zihaoxu20010109) [reply server/src/main/java/com/github/animelist/animelist/model/ratingsystem/RatingSystem.java]: left a comment regarding the use of epsilon and use absolute value for difference.

Comment (jvoucsb) [reply server/src/main/java/com/github/animelist/animelist/model/ratingsystem/RatingSystem.java]: We want the tolerance to only be on the “less than” side we want to guarantee the overall scores will n it exceed the max range. Absolute value allows a weight 1.01 to be valid thus we could have a score that is greater than the range.

**Review by rowantran — APPROVED** [review]
LGTM

**Review by AHuangHe — APPROVED** [review]
Looks Good!

**Pull request merged**